src/plugins/HUDPlugin.js

```diff
@@ -0,0 +1,87 @@
+import { BasePlugin } from './BasePlugin';
+
+/**
+ * @export
+ * @class HUDPlugin
+ * @extends {BasePlugin}
+ */
+export class HUDPlugin extends BasePlugin {
+  /**
+   * Creates an instance of HUDPlugin
+   * @param {object} params
+   * @param {string | HTMLElement} [params.positionsContainer] string or HTML Element that represents the container that the radio buttons should be added to
+   * @memberof HUDPlugin
+   */
+  constructor({ positionsContainer } = {}) {
+    super('HUD-Layout-Plugin');
+
+    this.positionControls =
+      positionsContainer instanceof HTMLElement
+        ? positionsContainer
+        : document.querySelector(positionsContainer);
+
+    this.radioButtons = [];
+    this.currentPos;
+  }
+
+  /**
+   * @memberof HUDPlugin
+   * @param {HTMLElement} pos the radio button that was clicked
+   */
+  onHUDToggle(pos) {
+    this.currentPos = pos.value;
+    pos.checked = true; //to ensure the radio button reflects its selected state to the user.
+    this.sendProperty(HUDPlugin.hudPositionKey, pos.value);
+  }
+
+  /**
+   * @memberof HUDPlugin
+   */
+  init() {
+    this.client.on(
+      'features',
+      function(features) {
+        if (!features.data || !features.data.hudPosition) {
+          return;
+        }
+        //get the game's reported HUD positions and build out the radio buttons
+        this.client.fetch('positions', result => {
```

Comment:
I'm wondering if we should name this less generically. `this.client.fetch('hudPositions', (result) => ...)`